compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -3093,4 +3093,120 @@ LogicalResult CustomOp::getResultTilePosition(
+    return iteratorTypes;
+  }
+  SmallVector<Range> getIterationDomain(Operation *op,
+                                        OpBuilder &builder) const;
+  FailureOr<TilingResult>
+  getTiledImplementation(Operation *op, OpBuilder &builder,
+                         ArrayRef<OpFoldResult> offsets,
+                         ArrayRef<OpFoldResult> sizes) const;
+
+  FailureOr<TilingResult>
+  generateResultTileValue(Operation *op, OpBuilder &builder,
+                          unsigned resultNumber, ArrayRef<OpFoldResult> offsets,
+                          ArrayRef<OpFoldResult> sizes) const {
+    return getTiledImplementation(op, builder, offsets, sizes);
+  }
+
+  LogicalResult
+  getResultTilePosition(Operation *op, OpBuilder &builder,
+                        unsigned resultNumber, ArrayRef<OpFoldResult> offsets,
+                        ArrayRef<OpFoldResult> sizes,
+                        SmallVector<OpFoldResult> &resultOffsets,
+                        SmallVector<OpFoldResult> &resultSizes) const {
+    resultOffsets.assign(offsets.begin(), offsets.end());
+    resultSizes.assign(sizes.begin(), sizes.end());
+    return success();
+  }
+};
+} // namespace
+
+SmallVector<Range>
+ConcatOpTilingExternalModel::getIterationDomain(Operation *op,
+                                                OpBuilder &builder) const {
+  auto concatOp = cast<tensor::ConcatOp>(op);
+  Location loc = concatOp.getLoc();
+  OpFoldResult zero = builder.getIndexAttr(0);
+  OpFoldResult one = builder.getIndexAttr(1);
+  SmallVector<Range> ranges;
+  for (auto dim : llvm::seq<int64_t>(0, concatOp.getResultType().getRank())) {
+    if (dim != concatOp.getDim()) {
+      OpFoldResult size = getDim(builder, loc, concatOp.getOperand(0), dim);
+      ranges.push_back(Range{zero, size, one});
+      continue;
+    }
+    SmallVector<AffineExpr> exprs(concatOp.getNumOperands());
+    bindSymbolsList(builder.getContext(), MutableArrayRef<AffineExpr>(exprs));
+    AffineExpr sum = exprs.front();
+    for (auto expr : llvm::drop_begin(exprs)) {
+      sum = sum + expr;
+    }
+    SmallVector<OpFoldResult> concatedDims =
+        llvm::map_to_vector(concatOp.getOperands(), [&](Value operand) {
+          return getDim(builder, loc, operand, dim);
+        });
+    OpFoldResult size =
+        affine::makeComposedFoldedAffineApply(builder, loc, sum, concatedDims);
+    ranges.push_back(Range{zero, size, one});
+  }
+
+  return ranges;
+}
+
+FailureOr<TilingResult> ConcatOpTilingExternalModel::getTiledImplementation(
+    Operation *op, OpBuilder &builder, ArrayRef<OpFoldResult> offsets,
+    ArrayRef<OpFoldResult> sizes) const {
+  auto concatOp = cast<tensor::ConcatOp>(op);
+  Location loc = concatOp.getLoc();
+  uint64_t dim = concatOp.getDim();
+  RankedTensorType resultType = concatOp.getResultType();
+
+  // Don't support tiling along the concated dimension.
+  std::optional<int64_t> concatDimSize = getConstantIntValue(sizes[dim]);
```

Comment:
This is a pessimization? This is disallowing dynamic tile sizes for all dimensions instead of just the concat dimension. I think you can just check that for the concat dimension, the tile size is 0.

Reply:
Lets talk about this more on a call. I am not sure I follow.